mac build broke again

I'm not exactly sure which commit (i can take a look when I get time) but the current head is broken for mac:

```
[100%] Linking C executable motion
Undefined symbols for architecture x86_64:
  "_avdevice_register_all", referenced from:
      _ffmpeg_global_init in ffmpeg.c.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [motion] Error 1
make[1]: *** [CMakeFiles/motion.dir/all] Error 2
make: *** [all] Error 2
```

Going back to here fixes it:
4cecdd5f03a798d53503f396d37611b0b626365b
```
git reset --hard 4cecdd5f03a798d53503f396d37611b0b626365b
```

Yeah, still doesn't work, it still believes pthread's isn't available:

relevant output:

```
checking pthread.h usability... yes
checking pthread.h presence... yes
checking for pthread.h... yes
checking pthread_np.h usability... no
checking pthread_np.h presence... no
checking for pthread_np.h... no
checking for threads... no
```


Relevant log:

```
configure:4139: checking pthread.h usability
configure:4139: gcc -c -g -O2 -I/usr/local/include -g -O2  -I/usr/local/include  conftest.c >&5
configure:4139: $? = 0
configure:4139: result: yes
configure:4139: checking pthread.h presence
configure:4139: gcc -E  -I/usr/local/include  conftest.c
configure:4139: $? = 0
configure:4139: result: yes
configure:4139: checking for pthread.h
configure:4139: result: yes
configure:4139: checking pthread_np.h usability
configure:4139: gcc -c -g -O2 -I/usr/local/include -g -O2  -I/usr/local/include  conftest.c >&5
conftest.c:59:10: fatal error: 'pthread_np.h' file not found
#include <pthread_np.h>
         ^
1 error generated.
configure:4139: $? = 1
configure: failed program was:

```

It looks like this commit broke it:
b5c3a73cc62e7641e550cf742a90a74f78bdae33



btw, I am using cmake to build:
git clone https://github.com/Motion-Project/motion.git
cd motion
cmake .

```
cmake .
-- The C compiler identification is AppleClang 8.0.0.8000042
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found PkgConfig: /usr/local/bin/pkg-config (found version "0.29.1") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE  
-- Found JPEG: /usr/local/lib/libjpeg.dylib  
-- Could NOT find MMAL (missing:  MMAL_LIBRARIES MMAL_INCLUDE_DIRS) 
-- Could NOT find MYSQL (missing:  MYSQL_LIBRARY MYSQL_INCLUDE_DIR) 
-- Could NOT find PostgreSQL (missing:  PostgreSQL_INCLUDE_DIR PostgreSQL_TYPE_INCLUDE_DIR) 
-- Checking for module 'sqlite3'
--   Found sqlite3, version 3.14.0
-- Looking for include file linux/videodev2.h
-- Looking for include file linux/videodev2.h - not found
-- Checking for modules 'libavutil;libavformat;libavcodec;libswscale'
--   Found libavutil, version 55.34.101
--   Found libavformat, version 57.56.101
--   Found libavcodec, version 57.64.101
--   Found libswscale, version 4.2.100
-- Configuration: 
--   Package Name:           motion
--   Package Version:        4.0.1+giteb137c5
--   Build System:           Darwin
--   Config Dir:             /usr/local/etc
--   Bin Dir:                /usr/local/bin
--   JPEG Include Dir:       /usr/local/include
--   WITH_V4L2:              ON
--   HAVE_V4L2:              OFF
--   WITH_BKTR:              OFF
--   HAVE_BKTR:              OFF
--   MYSQL:   OFF Include Dir: MYSQL_INCLUDE_DIR-NOTFOUND
--   SQLITE3: ON Include Dir: 
--   PGSQL:   OFF Include Dir: 
--   FFMPEG:  ON Include Dir: /usr/local/Cellar/ffmpeg/3.2.4/include
--   MMAL:    OFF
--   CFLAGS:   -std=gnu99 -Wall
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/mike/Projects/tempmotuon/motion
```

make

```
Scanning dependencies of target motion
[  4%] Building C object CMakeFiles/motion.dir/conf.c.o
[  8%] Building C object CMakeFiles/motion.dir/motion.c.o
[ 12%] Building C object CMakeFiles/motion.dir/alg.c.o
[ 16%] Building C object CMakeFiles/motion.dir/draw.c.o
[ 20%] Building C object CMakeFiles/motion.dir/event.c.o
[ 25%] Building C object CMakeFiles/motion.dir/ffmpeg.c.o
[ 29%] Building C object CMakeFiles/motion.dir/jpegutils.c.o
[ 33%] Building C object CMakeFiles/motion.dir/logger.c.o
[ 37%] Building C object CMakeFiles/motion.dir/md5.c.o
[ 41%] Building C object CMakeFiles/motion.dir/netcam.c.o
[ 45%] Building C object CMakeFiles/motion.dir/netcam_ftp.c.o
[ 50%] Building C object CMakeFiles/motion.dir/netcam_jpeg.c.o
[ 54%] Building C object CMakeFiles/motion.dir/netcam_rtsp.c.o
[ 58%] Building C object CMakeFiles/motion.dir/netcam_wget.c.o
[ 62%] Building C object CMakeFiles/motion.dir/picture.c.o
[ 66%] Building C object CMakeFiles/motion.dir/rotate.c.o
[ 70%] Building C object CMakeFiles/motion.dir/stream.c.o
[ 75%] Building C object CMakeFiles/motion.dir/track.c.o
[ 79%] Building C object CMakeFiles/motion.dir/video_loopback.c.o
[ 83%] Building C object CMakeFiles/motion.dir/webhttpd.c.o
[ 87%] Building C object CMakeFiles/motion.dir/video_v4l2.c.o
[ 91%] Building C object CMakeFiles/motion.dir/video_common.c.o
[ 95%] Building C object CMakeFiles/motion.dir/video_bktr.c.o
[100%] Linking C executable motion
Undefined symbols for architecture x86_64:
  "_avdevice_register_all", referenced from:
      _ffmpeg_global_init in ffmpeg.c.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [motion] Error 1
make[1]: *** [CMakeFiles/motion.dir/all] Error 2
make: *** [all] Error 2
```

if I do this then it works:
```
git reset --hard 042fd16d6cc7ecfaf002ae705605054b49f9e4f4
```

And just a note, when I go back to the previous commit here is the build log, I also tested the app and it works great:

```
-- Could NOT find MMAL (missing:  MMAL_LIBRARIES MMAL_INCLUDE_DIRS) 
-- Could NOT find MYSQL (missing:  MYSQL_LIBRARY MYSQL_INCLUDE_DIR) 
-- Could NOT find PostgreSQL (missing:  PostgreSQL_INCLUDE_DIR PostgreSQL_TYPE_INCLUDE_DIR) 
-- Checking for module 'sqlite3'
--   Found sqlite3, version 3.14.0
-- Checking for modules 'libavutil;libavformat;libavcodec;libswscale'
--   Found libavutil, version 55.34.101
--   Found libavformat, version 57.56.101
--   Found libavcodec, version 57.64.101
--   Found libswscale, version 4.2.100
-- Configuration: 
--   Package Name:           motion
--   Package Version:        4.0.1+git6ae6fd7
--   Build System:           Darwin
--   Config Dir:             /usr/local/etc
--   Bin Dir:                /usr/local/bin
--   JPEG Include Dir:       /usr/local/include
--   WITH_V4L2:              ON
--   HAVE_V4L2:              OFF
--   WITH_BKTR:              OFF
--   HAVE_BKTR:              OFF
--   MYSQL:   OFF Include Dir: MYSQL_INCLUDE_DIR-NOTFOUND
--   SQLITE3: ON Include Dir: 
--   PGSQL:   OFF Include Dir: 
--   FFMPEG:  ON Include Dir: /usr/local/Cellar/ffmpeg/3.2.4/include
--   MMAL:    OFF
--   CFLAGS:   -std=gnu99 -Wall
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/mike/Projects/tempmotuon/motion
Mikes-MacBook-Pro:motion mike$ make
Scanning dependencies of target motion
[  4%] Building C object CMakeFiles/motion.dir/conf.c.o
[  8%] Building C object CMakeFiles/motion.dir/motion.c.o
[ 12%] Building C object CMakeFiles/motion.dir/alg.c.o
[ 16%] Building C object CMakeFiles/motion.dir/draw.c.o
[ 20%] Building C object CMakeFiles/motion.dir/event.c.o
[ 25%] Building C object CMakeFiles/motion.dir/ffmpeg.c.o
/Users/mike/Projects/tempmotuon/motion/ffmpeg.c:304:23: warning: 'codec' is deprecated [-Wdeprecated-declarations]
        avcodec_close(AVSTREAM_CODEC_PTR(ffmpeg->video_st));
                      ^
/Users/mike/Projects/tempmotuon/motion/ffmpeg.c:31:47: note: expanded from macro 'AVSTREAM_CODEC_PTR'
#define AVSTREAM_CODEC_PTR(avs_ptr) (avs_ptr->codec)
                                              ^
/usr/local/include/libavformat/avformat.h:893:21: note: 'codec' has been explicitly marked deprecated here
    AVCodecContext *codec;
                    ^
/Users/mike/Projects/tempmotuon/motion/ffmpeg.c:343:17: warning: 'avcodec_encode_video2' is deprecated [-Wdeprecated-declarations]
        retcd = avcodec_encode_video2(AVSTREAM_CODEC_PTR(ffmpeg->video_st),
                ^
/usr/local/include/libavcodec/avcodec.h:5323:5: note: 'avcodec_encode_video2' has been explicitly marked deprecated here
int avcodec_encode_video2(AVCodecContext *avctx, AVPacket *avpkt,
    ^
/Users/mike/Projects/tempmotuon/motion/ffmpeg.c:343:39: warning: 'codec' is deprecated [-Wdeprecated-declarations]
        retcd = avcodec_encode_video2(AVSTREAM_CODEC_PTR(ffmpeg->video_st),
                                      ^
/Users/mike/Projects/tempmotuon/motion/ffmpeg.c:31:47: note: expanded from macro 'AVSTREAM_CODEC_PTR'
#define AVSTREAM_CODEC_PTR(avs_ptr) (avs_ptr->codec)
                                              ^
/usr/local/include/libavformat/avformat.h:893:21: note: 'codec' has been explicitly marked deprecated here
    AVCodecContext *codec;
                    ^
/Users/mike/Projects/tempmotuon/motion/ffmpeg.c:669:25: warning: 'codec' is deprecated [-Wdeprecated-declarations]
    ffmpeg->c     = c = AVSTREAM_CODEC_PTR(ffmpeg->video_st);
                        ^
/Users/mike/Projects/tempmotuon/motion/ffmpeg.c:31:47: note: expanded from macro 'AVSTREAM_CODEC_PTR'
#define AVSTREAM_CODEC_PTR(avs_ptr) (avs_ptr->codec)
                                              ^
/usr/local/include/libavformat/avformat.h:893:21: note: 'codec' has been explicitly marked deprecated here
    AVCodecContext *codec;
                    ^
/Users/mike/Projects/tempmotuon/motion/ffmpeg.c:861:23: warning: 'codec' is deprecated [-Wdeprecated-declarations]
        avcodec_close(AVSTREAM_CODEC_PTR(ffmpeg->video_st));
                      ^
/Users/mike/Projects/tempmotuon/motion/ffmpeg.c:31:47: note: expanded from macro 'AVSTREAM_CODEC_PTR'
#define AVSTREAM_CODEC_PTR(avs_ptr) (avs_ptr->codec)
                                              ^
/usr/local/include/libavformat/avformat.h:893:21: note: 'codec' has been explicitly marked deprecated here
    AVCodecContext *codec;
                    ^
5 warnings generated.
[ 29%] Building C object CMakeFiles/motion.dir/jpegutils.c.o
[ 33%] Building C object CMakeFiles/motion.dir/logger.c.o
[ 37%] Building C object CMakeFiles/motion.dir/netcam.c.o
[ 41%] Building C object CMakeFiles/motion.dir/netcam_ftp.c.o
[ 45%] Building C object CMakeFiles/motion.dir/netcam_jpeg.c.o
[ 50%] Building C object CMakeFiles/motion.dir/netcam_rtsp.c.o
/Users/mike/Projects/tempmotuon/motion/netcam_rtsp.c:102:13: warning: 'avcodec_decode_video2' is deprecated [-Wdeprecated-declarations]
    retcd = avcodec_decode_video2(cc, frame, &check, packet);
            ^
/usr/local/include/libavcodec/avcodec.h:4812:5: note: 'avcodec_decode_video2' has been explicitly marked deprecated here
int avcodec_decode_video2(AVCodecContext *avctx, AVFrame *picture,
    ^
/Users/mike/Projects/tempmotuon/motion/netcam_rtsp.c:160:19: warning: 'codec' is deprecated [-Wdeprecated-declarations]
    dec_ctx = st->codec;
                  ^
/usr/local/include/libavformat/avformat.h:893:21: note: 'codec' has been explicitly marked deprecated here
    AVCodecContext *codec;
                    ^
/Users/mike/Projects/tempmotuon/motion/netcam_rtsp.c:490:108: warning: 'codec' is deprecated [-Wdeprecated-declarations]
    netcam->rtsp->codec_context = netcam->rtsp->format_context->streams[netcam->rtsp->video_stream_index]->codec;
                                                                                                           ^
/usr/local/include/libavformat/avformat.h:893:21: note: 'codec' has been explicitly marked deprecated here
    AVCodecContext *codec;
                    ^
3 warnings generated.
[ 54%] Building C object CMakeFiles/motion.dir/netcam_wget.c.o
[ 58%] Building C object CMakeFiles/motion.dir/picture.c.o
[ 62%] Building C object CMakeFiles/motion.dir/rotate.c.o
[ 66%] Building C object CMakeFiles/motion.dir/stream.c.o
[ 70%] Building C object CMakeFiles/motion.dir/track.c.o
[ 75%] Building C object CMakeFiles/motion.dir/video_loopback.c.o
[ 79%] Building C object CMakeFiles/motion.dir/webhttpd.c.o
[ 83%] Building C object CMakeFiles/motion.dir/video_v4l2.c.o
[ 87%] Building C object CMakeFiles/motion.dir/video_common.c.o
[ 91%] Building C object CMakeFiles/motion.dir/video_bktr.c.o
[ 95%] Linking C executable motion
[100%] Built target motion
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -81,7 +81,7 @@ include_directories(${JPEG_INCLUDE_DIR})
 list(APPEND LINK_LIBRARIES ${CMAKE_THREAD_LIBS_INIT} ${JPEG_LIBRARIES})
 
 if(WITH_FFMPEG)
-	pkg_check_modules(FFMPEG REQUIRED libavutil libavformat libavcodec libswscale)
+	pkg_check_modules(FFMPEG REQUIRED libavutil libavformat libavcodec libswscale libavdevice)
 	include_directories(${FFMPEG_INCLUDE_DIRS})
 	link_directories(${FFMPEG_LIBRARY_DIRS})
 	list(APPEND LINK_LIBRARIES ${FFMPEG_LIBRARIES})

